Service monitor breaks the trivy operator

**What steps did you take and what happened:**
trivy operator version: `0.17.0-rc-arm64`
Sytem: raspberrypi 4 
Yaml configuration: 
```shell
operator:
  metricsVulnIdEnabled: true
  metricsConfigAuditInfo: true
  metricsRbacAssessmentInfo: true 
  metricsInfraAssessmentInfo: true
  metricsImageInfo: true
  metricsClusterComplianceInfo: true

trivy:
  ignoreUnfixed: true
  storageClassName: "openebs-hostpath"
  severity: HIGH,CRITICAL
trivyOperator:
  reportResourceLabels:
    source: trivy
nodeSelector:
  kubernetes.io/hostname: master
serviceMonitor:
  # -- enabled determines whether a serviceMonitor should be deployed
  enabled: true
  # -- The namespace where Prometheus expects to find service monitors

``` 
I got this error when enabling the service monitor
```shell
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                                               [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x23c6700]                                                                                                                                                                                                                                                                                                                                                                                                 goroutine 1465 [running]:                                                                                                                                                                                                             github.com/aquasecurity/trivy-operator/pkg/metrics.(*ResourcesMetricsCollector).collectClusterComplianceInfoReports(0x4002b81868, {0x3830100, 0x548ffe0}, 0x40000523a3?)                                                             ││     /home/runner/work/trivy-operator/trivy-operator/pkg/metrics/collector.go:1007 +0x1e0                                                                                                                                             ││ github.com/aquasecurity/trivy-operator/pkg/metrics.ResourcesMetricsCollector.Collect({{{0x3836b08, 0x40011e57a0}, 0x0}, {{0x40000523a3, 0xe}, {0x0, 0x0}, {0x0, 0x0}, {0x40000521f9, ...}, ...}, ...}, ...)                          ││     /home/runner/work/trivy-operator/trivy-operator/pkg/metrics/collector.go:583 +0x354                                                                                                                                              github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()                                                                                                                                                           /home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:455 +0xd8                                                                                                                             created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather in goroutine 1461                                                                                                                                        /home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:547 +0x974                                                                                                                            Stream closed EOF for trivy-operator/trivy-operator-545b4dcbb6-ztwsc (trivy-operator)     

``` 
**What did you expect to happen:**
Work normally 
**Anything else you would like to add:**
I tried the ``1.17.0.*` and `1.17.1.*` versions
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Trivy-Operator version `0.17.0-rc-arm64`
- Kubernetes version 1.28
- Raspberry pi 4

> @elyesbenamor @arifur-docker07 I think this issue has been fixed with `trivy-operator v0.18.0-rc` can you please check and confirm ?

i am using trivy-operator v0.18.0-rc now, however trivy-operator is still crashing-

Values.yaml changes are -

metricsFindingsEnabled: true
metricsVulnIdEnabled: true
exposedSecretScannerEnabled: true
metricsExposedSecretInfo: true
metricsConfigAuditInfo: true
metricsRbacAssessmentInfo: true
metricsInfraAssessmentInfo: true
metricsImageInfo: true
metricsClusterComplianceInfo: true
service:
  headless: false  # this is disable as per https://github.com/aquasecurity/trivy-operator/issues/1098 for metrics visualization
serviceMonitor:
    enabled: false
---
Below logs are when Service Monitor is enable-
---
{"level":"info","ts":"2024-01-02T07:41:49Z","msg":"Starting workers","controller":"customresourcedefinition","controllerGroup":"apiextensions.k8s.io","controllerKind":"CustomResourceDefinition","worker count":1}
{"level":"info","ts":"2024-01-02T07:41:49Z","msg":"Starting workers","controller":"configmap","controllerGroup":"","controllerKind":"ConfigMap","worker count":1}
{"level":"info","ts":"2024-01-02T07:41:49Z","msg":"Starting workers","controller":"replicationcontroller","controllerGroup":"","controllerKind":"ReplicationController","worker count":1}
{"level":"info","ts":"2024-01-02T07:41:49Z","msg":"Starting workers","controller":"configmap","controllerGroup":"","controllerKind":"ConfigMap","worker count":1}
I0102 07:41:50.070608       1 request.go:697] Waited for 1.0029985s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/api/v1/namespaces/trivy-system/secrets/trivy-operator-trivy-config
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x31fcd19]

goroutine 6170 [running]:
github.com/aquasecurity/trivy-operator/pkg/metrics.(*ResourcesMetricsCollector).collectClusterComplianceInfoReports(0xc00271f878, {0x4bc6700, 0x6d7fe60}, 0xc00007a033?)
        /home/runner/work/trivy-operator/trivy-operator/pkg/metrics/collector.go:1007 +0x239
github.com/aquasecurity/trivy-operator/pkg/metrics.ResourcesMetricsCollector.Collect({{{0x4bcffc8, 0xc0017869c0}, 0x0}, {{0xc00007a033, 0xc}, {0x0, 0x0}, {0x0, 0x0}, {0xc000078469, ...}, ...}, ...}, ...)
        /home/runner/work/trivy-operator/trivy-operator/pkg/metrics/collector.go:583 +0x478
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
        /home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:455 +0x102
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather in goroutine 6166
        /home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:547 +0xbab
---
Below logs are when Service monitor is disable-
{"level":"info","ts":"2024-01-02T07:48:45Z","msg":"Starting workers","controller":"configmap","controllerGroup":"","controllerKind":"ConfigMap","worker count":1}
{"level":"info","ts":"2024-01-02T07:48:45Z","msg":"Starting workers","controller":"networkpolicy","controllerGroup":"networking.k8s.io","controllerKind":"NetworkPolicy","worker count":1}
{"level":"info","ts":"2024-01-02T07:48:45Z","msg":"Starting workers","controller":"configmap","controllerGroup":"","controllerKind":"ConfigMap","worker count":1}
I0102 07:48:46.950363       1 request.go:697] Waited for 1.002802144s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/api/v1/namespaces/trivy-system/secrets/trivy-operator-trivy-config
{"level":"error","ts":"2024-01-02T07:48:47Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-9977dc9b4","container":"odb-tools","status.reason":"OOMKilled","status.message":"Killed\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:346\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).SetupWithManager.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:81\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/reconcile/reconcile.go:111\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x31fcd19]

goroutine 2005 [running]:
github.com/aquasecurity/trivy-operator/pkg/metrics.(*ResourcesMetricsCollector).collectClusterComplianceInfoReports(0xc001cad878, {0x4bc6700, 0x6d7fe60}, 0xc00007c133?)
        /home/runner/work/trivy-operator/trivy-operator/pkg/metrics/collector.go:1007 +0x239
github.com/aquasecurity/trivy-operator/pkg/metrics.ResourcesMetricsCollector.Collect({{{0x4bcffc8, 0xc0012ce300}, 0x0}, {{0xc00007c133, 0xc}, {0x0, 0x0}, {0x0, 0x0}, {0xc000078049, ...}, ...}, ...}, ...)
        /home/runner/work/trivy-operator/trivy-operator/pkg/metrics/collector.go:583 +0x478
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
        /home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:455 +0x102
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather in goroutine 2001
        /home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:547 +0xbab

--- a/pkg/metrics/collector.go
+++ b/pkg/metrics/collector.go
@@ -1004,25 +1004,27 @@ func (c *ResourcesMetricsCollector) collectClusterComplianceInfoReports(ctx cont
 		if c.Config.MetricsClusterComplianceInfo {
 			labelValues[0] = r.Spec.Complaince.Title
 			labelValues[1] = r.Spec.Complaince.Description
-			for _, summary := range r.Status.SummaryReport.SummaryControls {
-				if summary.TotalFail == nil {
-					continue
-				}
-				status := PassStatus
-				metricCounter := 1
-				if *summary.TotalFail > 0 {
-					status = FailStatus
-					metricCounter = *summary.TotalFail
-				}
-				labelValues[2] = summary.ID
-				labelValues[3] = summary.Name
-				labelValues[4] = NewStatusLabel(Status(status)).Label
-				labelValues[5] = summary.Severity
+			if r.Status.SummaryReport != nil {
+				for _, summary := range r.Status.SummaryReport.SummaryControls {
+					if summary.TotalFail == nil {
+						continue
+					}
+					status := PassStatus
+					metricCounter := 1
+					if *summary.TotalFail > 0 {
+						status = FailStatus
+						metricCounter = *summary.TotalFail
+					}
+					labelValues[2] = summary.ID
+					labelValues[3] = summary.Name
+					labelValues[4] = NewStatusLabel(Status(status)).Label
+					labelValues[5] = summary.Severity
 
-				for i, label := range c.GetReportResourceLabels() {
-					labelValues[i+6] = r.Labels[label]
+					for i, label := range c.GetReportResourceLabels() {
+						labelValues[i+6] = r.Labels[label]
+					}
+					metrics <- prometheus.MustNewConstMetric(c.complianceInfoDesc, prometheus.GaugeValue, float64(metricCounter), labelValues...)
 				}
-				metrics <- prometheus.MustNewConstMetric(c.complianceInfoDesc, prometheus.GaugeValue, float64(metricCounter), labelValues...)
 			}
 		}
 	}

